### PR TITLE
Provide only createContainer API (BREAKING CHANGE)

### LINKED ISSUE #12: RFC: new API design with breaking changes
There's some drawbacks in the current design and I would like to redesign the API.
This is a good moment to get feedback because this is going to be a breaking change.

## Problem 1:
`<Provider useValue={...}>` takes a hook as a prop. This is not very good because props may change. Currently, to mitigate the problem, it warns if useValue is changed between renders.

#### My proposal for problem 1:
We only accept useReduder style state.
`<Provider reducer={...} initialState={...} init={...}>`

## Problem 2:
`createContainer(() => useReducer(...))` doesn't allow receiving initial value from props.

#### My proposal for problem 2:
Make it like `createContainer((props) => ...)`, and accept `props` from `<Provider {...props}>`

----

I would like to hear +1, -1, comments, suggestions, other problems, and so on. Thanks for your contribution!

### PR DESCRIPTION
**Pull request by dai-shi**
Hopefully, this closes #12, #13 and #14.

This removes default context APIs completely and only keeps `createContainer` API.
This may not fulfill everyone's needs, but it should provide a primitive so that they can easily adopt it to their use cases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,18 +134,20 @@ const {
...
 ```
 
-### useDispatch
+### useUpdate
 
 ```javascript
 const Component = () => {
-  const dispatch = useDispatch(); // simply to get the second one of the tuple
+  // this hook returns the second one of the tuple returned by useValue.
+  // it can be renamed as `dispatch`, `setState`, `actions`, or anything.
```
Comment (dai-shi) [inline README.md]: I wonder if it makes sense to everyone.

**Comment by dai-shi** [comment]
Please check out README and examples.
Your feedback is pretty much appreciated.
@Wildhoney @kevinya @ScriptedAlchemy

**Comment by ScriptedAlchemy** [comment]
Checking 👍

**Comment by ScriptedAlchemy** [comment]
Readme looks simpler now, i prefer the new API - with your `useReducer (props)` and embedded example. It might be worth adding a little description text explaining the "basics" of whats going on under the titles. 

I'm just thinking about this aspect. People find redux hard and hooks scary, for wider developer react. Under the title, a small chunk of text on a) what this is doing b) why or when it would be useful.

The recipes are extremely helpful to me. But id likely has to explain the "ingredients" to many many developers. 

```
const {
  Provider,
  useTracked,
  // ...
} = createContainer(({ reducer, initialState, init }) => useReducer(reducer, initialState, init));```

"We create the container and pass a reducer directly to it so that the provider and hook come with context easily wired up" or something like that (probably not accurate lol) 

Trying to cover all bases and ive been pushing this project and reactive redux as alternatives for the global state, prop drilling or avoiding a whole redux implementation for some simple global access thats saner than vanilla react.

A big problem my projects have had is either too much verbosity or not enough explanation around some of the code samples. Even if you mentioned how both work elsewhere, a slight re-iteration and even a link that jumps to the functions more fully fleshed implementation documentation does help.

admittedly ive not looked in depth at your examples, but i am thankful for them.

One thing that would be a little more time consuming but super helpful is something other than the typical increment-deincrement examples. Nothing against your examples, this is just a general beef i have noted and experienced from time to time. It's kinda hard to see the real-world example.

If you ever end up building some kind of 3 page mini-site - it can be helpful.
https://github.com/scriptedalchemy/redux-first-router-demo

This demo has literally been the starting point for companies, ive come into help on a contract basis and its a clone of the demo. It helps with adoption. Just something to think about, the same demo site could be retrofitted over and over for all your various projects and would give a real world value example right away.

an example, your service worker hook. I love it and know how to use it, starting to think about real world examples. If you had a boilerplate demo-site (that could serve as a "hire me" site as well lol) things like your worker plugin, being used for trivial things like maybe looping over something would connect the dots with more developers.

**Comment by dai-shi** [comment]
> adding a little description text explaining the "basics"

Will try.

> A big problem my projects have had

Is this mainly about the README? Yeah, I think it should be improved. Admittedly, that's something I'm not super good at.

> the real-world example.

That would be great for sure. It may take time, but I'll think about what kind of demo would be good. Any suggestions are welcome.

**Comment by dai-shi** [comment]
I wish I had more feedback.
Let me merge this and publish this breaking change to possibly get more feedback...

**Comment by dai-shi** [comment]
Published: https://www.npmjs.com/package/react-tracked/v/0.8.0

**Comment by ScriptedAlchemy** [comment]
I’m not great with readmes either, I’ll also try and gather feedback when I’ve got these things implemented. 

To be honest, I might not have released a whole lot in the past few months, but when I do breaking changes on my large libraries. I’d just let the PR sit and tag some users. Then merge and fast follow if any issues opened. I’ll think on some examples for a demo site. It’s  not the fastest for sure. But it helps developing tools and testing them in your own playground

**Comment by dai-shi** [comment]
I see what you mean. (Though, I'm afraid this lib is not yet that large... not in size, but in popularity.)

To tell the truth, I've been thinking about this change for months.
Just recently, several people reported some issues, so it was a good timing for me.

Anyway, thanks a lot for your comment.

**Comment by ScriptedAlchemy** [comment]
Thank you for your work! It’s some of the best and most forward thinking code I’ve seen. I appreciate your efforts and such a refreshing look on react implementations.

**Pull request merged**